packages/core/src/guided-answers-api.ts

```diff
@@ -171,13 +172,41 @@ async function getTrees(host: string, queryOptions?: GuidedAnswersQueryOptions):
     const query = queryOptions?.query ? encodeURIComponent(`"${queryOptions.query}"`) : '*';
     const urlGetParamString = convertQueryOptionsToGetParams(queryOptions?.filters, queryOptions?.paging);
     const url = `${host}${TREE_PATH}${query}${urlGetParamString}`;
-    const response: AxiosResponse<GuidedAnswerTreeSearchResult> = await axios.get<GuidedAnswerTreeSearchResult>(url);
-    const searchResult = response.data;
-    if (!Array.isArray(searchResult?.trees)) {
-        throw Error(
-            `Query result from call '${url}' does not contain property 'trees' as array. Received response: '${searchResult}'`
-        );
+
+    // Cancel the previous request if it exists
+    if (previousToken.length) {
+        const prev = previousToken.pop();
+        prev?.cancel('Canceling previous request');
     }
+
+    // Create a new CancelToken for the current request
+    const source = axios.CancelToken.source();
+    previousToken.push(source);
+
+    let searchResult: GuidedAnswerTreeSearchResult = {
+        resultSize: -1,
+        trees: [],
+        productFilters: [],
+        componentFilters: []
+    };
+    await axios
+        .get<GuidedAnswerTreeSearchResult>(url, {
+            cancelToken: source.token
+        })
+        .then((response) => {
+            searchResult = response.data;
+            if (!Array.isArray(searchResult?.trees)) {
+                throw Error(`Query result from call '${url}' does not contain property 'trees' as array`);
+            }
+        })
+        .catch((thrown) => {
+            if (axios.isCancel(thrown)) {
+                console.log(`Request canceled: '${thrown.message}'`);
+            } else {
+                throw Error(`Error fetching selection: '${thrown.message}'`);
+            }
+        });
+
```

Comment:
Why are you mixing `await` with `.then` and `.catch`. If we are using await we can directly assign the return value and embed it into throw/catch, which is consistent with the rest of the code (and easier to read I think).
Also, please pass in the logger and use it to log the cancel event.
Suggestion:

```suggestion
    try {
        const response = await axios.get<GuidedAnswerTreeSearchResult>(url, {
            cancelToken: source.token
        });

        searchResult = response.data;
        if (!Array.isArray(searchResult?.trees)) {
            throw Error(`Query result from call '${url}' does not contain property 'trees' as array`);
        }
    } catch (error) {
        if (axios.isCancel(error)) {
            logger.logDebug(`Request canceled: '${error.message}'`);
        } else {
            throw error;
        }
    }
```

You'd also need to change the signature of this function to 
```typescript
async function getTrees(
    host: string,
    logger: Logger,
    queryOptions?: GuidedAnswersQueryOptions
): Promise<GuidedAnswerTreeSearchResult> 
```

adjust the JSDoc and pass the logger into it.

Reply:
Done thanks